src/xdg.c

```diff
@@ -48,21 +48,19 @@ handle_new_xdg_popup(struct wl_listener *listener, void *data)
 static bool
 has_ssd(struct view *view)
 {
-	if (!rc.xdg_shell_server_side_deco) {
-		return false;
-	}
-
-	/*
-	 * Some XDG shells refuse to disable CSD in which case their
-	 * geometry.{x,y} seems to be greater than zero. We filter on that
-	 * on the assumption that this will remain true.
-	 */
-	struct wlr_xdg_surface_state *current =
-		&xdg_surface_from_view(view)->current;
-	if (current->geometry.x || current->geometry.y) {
+	/* view->ssd_preference is set by the kde or xdg decoration variant */
```

Comment:
Spell out `xdg-decoration` or `kde-server-decoration`

Reply:
I've used the `.c` files instead but happy to change to the protocol names if desired.